lib/src/internet_connection.dart

```diff
@@ -127,11 +127,19 @@ class InternetConnection {
     InternetCheckOption option,
   ) async {
     try {
-      final response = await http.head(option.uri).timeout(option.timeout);
+      final response = await http
+          .head(
+            option.uri,
+            headers: option.headers,
+          )
+          .timeout(option.timeout);
+
+      ResponseStatusFn statusFn =
+          option.responseStatusFn ?? (response) => response.statusCode == 200;
```

Comment:
Please change this to:

```dart
defaultResponseStatusFn(response) => response.statusCode == 200;
final statusFn = option.responseStatusFn ?? defaultResponseStatusFn;
```

Reply:
I've assign default responseStatusFn on the constructor, so we don't need tho define it here